sdk/storage/Azure.Storage.DataMovement/src/StorageTransferManager.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+// To use AsyncQueue, might decide to go with another queue later
+using Microsoft.VisualStudio.Threading;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+using System.Threading;
+using Azure.Storage.Blobs.Models;
+using System.Threading.Tasks;
+using System.IO;
+
+namespace Azure.Storage.DataMovement
+{
+    /// <summary>
+    /// TransferManager class
+    /// </summary>
+    public class StorageTransferManager
+    {
+        // To hold the jobs to scan
+        // This is a weird thing to have because we have regular Blob Directory Upload / Download which will
+        // also call the scanner on it's own. Something to think about is whehter or not doing scanning in a separate
+        // part of DMLib instead of scanning right before the job is benefical.
+        private AsyncQueue<StorageTransferJob> _toScanQueue;
+        // To hold the jobs that have finished scanning and ready to run; This will help with grabbing required
+        // authentication from the original job and for updating the jobs for progress tracking
+        private AsyncQueue<StorageTransferJob> _jobsToProcess;
+        // Not sure if we should keep the jobs that in in progress here
+        //private IList<StorageTransferJob> _jobsInProgress;
+        // local directory path to put hte memory mapped file of the progress tracking. if we pause or break
+        // we will have the information on where to continue from.
+        private string _progressLogDirectoryPath;
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="progressLogDirectoryPath">Specify directory path to log file to continue from.</param>
+        public StorageTransferManager(string progressLogDirectoryPath = default)
```

Comment:
So that's the path where DMLib maintains state? We should find better name for this.
I don't think we should call out in docs what exactly, we just need vague statement like "directory where transfer state is kept".

Reply:
Updated.